### PR TITLE
Fixed billable error on the timesheet entries

### PR DESCRIPTION
**Pull request by prasanthchaduvula**
## Notion card
https://www.notion.so/6afc9500b2cc42af86be55f37894fb09?v=185bd45b9dc24dfcad9c57c07d32a7c0&p=719807ecfcce41d99860f66c04159422&pm=s

## Summary

- Updated the Badge UI of billable tags like billed, non billable and unbilled.
- Fixed In edit mode, for the same timesheet entry the billable checkbox should remain unchecked . The user can again make the entry as billable. 

## Preview

https://www.loom.com/share/01d591e011794030a2f379583178df04

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Comment by prasanthchaduvula** [comment]
@apoorv1316 Flaky test on cypress https://github.com/saeloun/miru-web/actions/runs/3092832700/jobs/5004546339
I am not sure about it. Please have a look

Please test the PR app regarding the bug fix.

**Review by swaathi — APPROVED** [review]

**Comment by prasanthchaduvula** [comment]
@akhilgkrishnan I created Badge reusable styled component. I want to update `alias.js` file and I tried but config not working. As of now everything working fine but it's better if we add it to `alias.js` file for better imports in future.

**Comment by supriya3105** [comment]
@akhilgkrishnan Can you review ?

**Review by apoorv1316 — APPROVED** [review]

File: app/javascript/src/styledComponents/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+
+import classnames from "classnames";
+import PropTypes from "prop-types";
```
Comment (akhilgkrishnan) [inline app/javascript/src/styledComponents/Badge.tsx:4]: You don't need a proptype checking if you are using typescript, you can directly use the typescript one.

```suggestion
type Props = {
  color: string;
};
```

File: app/javascript/src/styledComponents/Badge.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+const DEFAULT_STYLE = "inline-flex items-center rounded-full px-2.5 py-0.5 text-xs font-semibold leading-4 tracking-wider text-purple-800 bg-purple-100";
+
+const Badge = ({
+  text = null,
+  color = "",
+  bgColor = "",
+  className = ""
+}) => (
```
Comment (akhilgkrishnan) [inline app/javascript/src/styledComponents/Badge.tsx:13]:
```suggestion
const Badge = ({
  text = null,
  color = "",
  bgColor = "",
  className = ""
}:  Props) => (
```

**Review by akhilgkrishnan — CHANGES_REQUESTED** [review]
@prasanthchaduvula Added some changes. Rest LGTM 👍

**Review by akhilgkrishnan — APPROVED** [review]
LGTM

**Pull request merged**